packages/core/types/src/types/core/common/middleware.ts

```diff
@@ -6,6 +6,14 @@ export type MiddlewareFactory<T = any> = (
   ctx: { strapi: Strapi }
 ) => MiddlewareHandler | void;
 
+export type MiddlewareName = string;
+
+export type MiddlewareConfig = {
+  name?: string;
```

Comment:
Well, basically everywhere we have a name, it could be a middleware UID instead I believe

Reply:
This one needs to work for user custom module names which can be any string, but I can at least do MiddlewareName | string so that autosuggest may work.